### PR TITLE
Add simulation with all diagnostics turned to integration testing

### PR DESCRIPTION
**Pull request by msulprizio**
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard

### Describe the update

In order to evaluate all diagnostic collections, we now include a simulation with all collections activated in HISTORY.rc. This is currently a copy of the `gc_4x5_merra2_fullchem_benchmark` simulation with the comment character removed from all collection names at the top of `HISTORY.rc` with the name `gc_4x5_merra2_fullchem_alldiags`. 

Still to do:

- [x] Add all diagnostics simulation to GCHP integration tests
- [x] Turn on ObsPack and Planeflight in `geoschem_config.yml` for GCClassic test
- [x] Fix display of superproject and submodule commit hashes in log files and to stdout
 
### Expected changes

This is a zero diff update with respect to benchmark simulations.

### Related Github Issue

- https://github.com/geoschem/geos-chem/issues/2143
- https://github.com/geoschem/geos-chem/issues/714

### CODE REVIEW

**Comment by msulprizio** [comment]
The all diagnostics run is currently failing integration tests:

```
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #c49fcec Submod updates: Merge GEOS-Chem PR #2353 and Cloud-J PR #19
GEOS-Chem #2c8910869 Add ModelE2.1 (GCAP 2.0) simulation to GCClassic integration tests
HEMCO     #2192e0e HEMCO 3.9.1 release

Using 24 OpenMP threads
Number of execution tests: 30

Submitted as SLURM job: 41196464
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gc_05x0625_NA_47L_merra2_CH4........................Execute Simulation....PASS
gc_05x0625_NA_47L_merra2_fullchem...................Execute Simulation....PASS
gc_2x25_ModelE2.1_fullchem..........................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem..........................Execute Simulation....PASS
gc_4x5_47L_merra2_fullchem_TOMAS15..................Execute Simulation....PASS
gc_4x5_merra2_aerosol...............................Execute Simulation....PASS
gc_4x5_merra2_carbon................................Execute Simulation....PASS
gc_4x5_merra2_carbon_CH4............................Execute Simulation....PASS
gc_4x5_merra2_carbon_CO.............................Execute Simulation....PASS
gc_4x5_merra2_carbon_CO2............................Execute Simulation....PASS
gc_4x5_merra2_carbon_OCS............................Execute Simulation....PASS
gc_4x5_merra2_CH4...................................Execute Simulation....PASS
gc_4x5_merra2_CO2...................................Execute Simulation....PASS
gc_4x5_merra2_fullchem..............................Execute Simulation....PASS
gc_4x5_merra2_fullchem_aciduptake...................Execute Simulation....PASS
gc_4x5_merra2_fullchem_alldiags.....................Execute Simulation....FAIL
...
```

The error in the log file is:

```
===============================================================================
GEOS-Chem ERROR: The value of attribute "UVFlux.template:" must end with a 
single quote character, followed by a comma. Please check the HISTORY.rc file.
 -> at Init_DiagList (Headers/diaglist_mod.F90)

 -> ERROR occurred at (or near) line    397 of the HISTORY.rc file
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Init_DiagList"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
 Timer forced to stop due to error: GEOS-Chem                     
 Timer forced to stop due to error: Diagnostics                   
```

We will investigate and add fixes to this branch.

**Comment by yantosca** [comment]
The GCHP "alldiags" integration test fails with this error:
```console
GEOS-Chem ERROR [0006]: The mapData%slot2Id array corresponding to collection "BudgetWetDepFull" contains missing values! This can indicate that this collection is either undefined or turned off.  Please check the HISTORY.rc configuration file in your run directory.
 --> LOCATION:  -> at Get_Mapping (in module Headers/state_diag_mod.F90)
 
GEOS-Chem ERROR [0006]: Error encountered in "Get_Mapping": BudgetWetDepFull
 --> LOCATION:  -> at Get_MapData_and_NumSlots (in module Headers/state_diag_mod.F90)
 
GEOS-Chem ERROR [0006]: Error encountered in "Get_MapData_and_NumSlots"!
 --> LOCATION:  -> at Init_and_Register_R8_3D (in module Headers/state_diag_mod.F90)
 
GEOS-Chem ERROR [0006]: Error encountered in "Init_and_Register", diagID =BudgetWetDepFull
 --> LOCATION:  -> at Init_State_Diag (in Headers/state_diag_mod.F90)
 
GEOS-Chem ERROR [0006]: Error encountered within call to "Init_State_Diag"!
 --> LOCATION:  -> at GC_Init_StateObj (in GeosCore/gc_environment_mod.F90)
pe=00006 FAIL at line=00255    gchp_chunk_mod.F90                       <Error calling GC_Init_StateObj>
pe=00006 FAIL at line=01498    Chem_GridCompMod.F90                     <status=1>
pe=00006 FAIL at line=01807    MAPL_Generic.F90                         <status=1>
```
We are investigating further.

**Comment by yantosca** [comment]
The GCHP error is due to the fact that the budget diagnostic used O3 as the sample species in the `HISTORY.rc.fullchem` template file:
```kconfig
 Budget.fields:          'BudgetWetDepFull_O3          ', 'GCHPchem',
                          'BudgetChemistryFull_O3       ', 'GCHPchem',
                          'BudgetConvectionFull_O3      ', 'GCHPchem',
                          'BudgetMixingFull_O3          ', 'GCHPchem',
                          'BudgetEmisDryDepFull_O3      ', 'GCHPchem',
                          #
                          'BudgetWetDepLevs1to35_O3     ', 'GCHPchem',
                          'BudgetChemistryLevs1to35_O3  ', 'GCHPchem',
                          'BudgetConvectionLevs1to35_O3 ', 'GCHPchem',
                          'BudgetMixingLevs1to35_O3     ', 'GCHPchem',
                          'BudgetEmisDryDepLevs1to35_O3 ', 'GCHPchem',
                           #
                          'BudgetWetDepTrop_O3          ', 'GCHPchem',
                          'BudgetChemistryTrop_O3       ', 'GCHPchem',
                          'BudgetConvectionTrop_O3      ', 'GCHPchem',
                          'BudgetMixingTrop_O3          ', 'GCHPchem',
                          'BudgetEmisDryDepTrop_O3      ', 'GCHPchem',
                           #
                          'BudgetWetDepPBL_O3           ', 'GCHPchem',
                          'BudgetChemistryPBL_O3        ', 'GCHPchem',
                          'BudgetConvectionPBL_O3       ', 'GCHPchem',
                          'BudgetMixingPBL_O3           ', 'GCHPchem',
                          'BudgetEmisDryDepPBL_O3  
```
But O3 is not water-soluble, and thus does not have a wet deposition ID.  The `BudgetWetDep*_O3` fields are therefore undefined, and caused GCHP to exit with an error.

We have now changed the sample species from O3 to SO4, which is water soluble.

**Comment by yantosca** [comment]
Another GCHP error was revealed by the "alldiags" test:
```console
GEOS-Chem ERROR [0010]: Could not find bin index for tag name: 574NM
 --> LOCATION:  -> at Get_UVFlux_Index (in module Headers/state_diag_mod.F90)
 
GEOS-Chem ERROR [0010]: Error encountered in routine "Get_UVFlux_Bin"!
 --> LOCATION:  -> at Get_Mapping (in module Headers/state_diag_mod.F90)
```
@lizziel and I have traced it to routine `Get_UVFlux_Bin` in `Headers/state_diag_mod.F90`:
```F90
    ! Get the index for the tagname                                                       
    DO N = 1, 18
       IF ( TRIM( tagName ) == TRIM( UVFlux_Tag_Names(N) ) ) THEN
          bin = N
          EXIT
       ENDIF
    ENDDO

    ! Trap potential errros                                                               
    IF ( bin < 0 ) THEN
       errMsg = 'Could not find bin index for tag name: ' // TRIM( tagName )
       CALL GC_Error( errMsg, RC, thisLoc )
       RETURN
    ENDIF
```
The error is that tagName is has "NM" in uppercase whereas `TRIM( UVFlux_Tag_Names(N)` has `nm` in lowercase.  We will make sure that both are in uppercase when we do the comparison and rerun the tests.

**Comment by yantosca** [comment]
Encountered yet another error in the GCHP alldiags test.  I think the job ran out of memory.  I bumped up the requested memory to 450000 GB and will see if that goes

**Comment by yantosca** [comment]
The issue was not the job running out of memory, but the StratBM diagnostic entries were commented out.  I've uncommented those entries but have disabled the StratBM collection for the GCHP alldiags test, as that is only needed
for 10-yr benchmarks.

**Comment by yantosca** [comment]
Found that the GCHP DryDep collection entries had a bunch of missing `_` characters, so entries were e.g. `DryDepACET` instead of `DryDep_ACET`.  This was causing the GCHP alldiags test to fail.  A fix has been pushed in commit dcc8809 and the GCHP integration tests are re-running.

**Comment by yantosca** [comment]
Also pushed the following updates:
1. Fixed several inconsistencies in the emissions diagnostics between `HISTORY.rc` and `HEMCO_Diagn.rc` for GCHP.
2. Disable the `UVFlux` diagnostic in the "alldiags" tests.  Turning this on in GCHP results in a floating-point error, see: https://github.com/geoschem/GCHP/issues/436

**Comment by yantosca** [comment]
After adding all of the above-mentioned fixes, all GCHP integration tests now pass:
```console
==============================================================================
GCHP: Execution Test Results

CodeDir       : 2d1bc44 GEOS-Chem submod update: Merge PR #2394 (Add GCAP int test)
MAPL          : 231d53c Merge pull request #36 from geoschem/feature/improve_hflux_regridding
GMAO_Shared   : 4ddb3ec Merge pull request #2 from geoschem/feature/mapl-upgrade
ESMA_cmake    : ad5deba Added ecbuild as a submodule of ESMA_cmake
gFTL-shared   : 4b82492 Merge branch 'upstream_v1.5.0' into feature/v1.5.0
FMS           : 259759d Merge pull request #3 from geoschem/feature/update_gmao_libs
FVdycoreCubed : af42462 Merge PR #8 (Add PLEadv diagnostic for offline advection in GCHP)
geos-chem     : ee5a477e6 Add an extra "#" to commented-out emission diagnostics
HEMCO         : e68ebae HEMCO 3.9.2 release
yaFyaml       : 19afe50 Merge branch 'upstream_v1.0.4' into feature/v1.0.4
pFlogger      : 2c4b724 Merge branch 'upstream_v1.9.1' into feature/v1.9.1
Cloud-J       : 37a4ece Update release version number and date
HETP          : 2a99b24 Merge pull request #2 from geoschem/bugfix/initialize_local_variables

Number of execution tests: 12

Submitted as SLURM job: 42064039
==============================================================================
...
gchp_merra2_fullchem_alldiags.......................Execute Simulation....PASS
...
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
All tests were zero-diff w/r/t PR #2394.

**Comment by yantosca** [comment]
@msulprizio: GitHub wouldn't let me add you as a reviewer since this was originally your PR.  But please do feel free to submit a review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,23 +8,32 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Added
 - Added number of levels with clouds for photolysis to geoschem_config.yml and Input_Opt to pass to Cloud-J
 - Added `State_Grid%CPU_Subdomain_ID` and `State_Grid%CPU_Subdomain_FirstID` as "identifier numbers" for multiple instances of GEOS-Chem on one core in WRF and CESM
+- Added simulation with all diagnostics on in HISTORY.rc to integration tests for GCClassic (including Planeflight + ObsPack)
```
Comment (msulprizio) [inline CHANGELOG.md:11]: Mention GCHP all diags test here too

**Review by msulprizio — COMMENTED** [review]
This looks good to me @yantosca. Thanks for resolving these issues! I have one minor comment in the CHANGELOG but pending that this is good for merging.

**Review by yantosca — APPROVED** [review]

**Comment by yantosca** [comment]
@lizziel @msulprizio: I merged (locally) into dev/no-diff-to-benchmark and reran integration tests.  All tests passed and were zero-diff w/r/t the prior PR (#2394).

**Pull request merged**